planner/core/rule_partition_processor.go

```diff
@@ -758,6 +758,9 @@ func makePartitionByFnCol(sctx sessionctx.Context, columns []*expression.Column,
 		fn = raw
 		args := fn.GetArgs()
 		if len(args) > 0 {
+			if expression.ExtractColumnSet(args).Len() > 1 {
```

Comment:
This may be not quite true.
In `add(col, const)`,  it's still a monotonous function, but in `add(col1, col2)`, this one is not.
And the code is not quite readable because the reader (or as a reviewer, me) does not know why without knowing issue #22898 

Maybe we can move line 769 to 759 to check monotonous immediately?
```
fn = raw
monotonous = getMonotoneMode(raw.FuncName.L)
if monotonous == monotoneModeInvalid {
    return ...
}
```

Reply:
I have moved it, PTAL.